go_test.go

```diff
@@ -0,0 +1,171 @@
+			},
+			error(nil),
+		},
+		{ // 2
+			[]byte(fileB),
+			[]byte(fileBRename),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 0, DelLines: 3, Inserted: false, Deleted: true, IsOther: false},
+				&Change{Name: "func xxxx", InsLines: 3, DelLines: 0, Inserted: true, Deleted: false, IsOther: false},
+			},
+			error(nil),
+		},
+		{ // 3
+			[]byte(fileB),
+			[]byte(fileBDelete),
+			[]*Change{
+				&Change{Name: "func bbbb", InsLines: 0, DelLines: 3, Inserted: false, Deleted: true, IsOther: false},
+				&Change{Name: "other", InsLines: 0, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 4
+			[]byte(fileA),
+			[]byte(fileBRename),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 0, DelLines: 1, Inserted: false, Deleted: true, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func xxxx", InsLines: 3, DelLines: 0, Inserted: true, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 2, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 5
+			[]byte(fileA),
+			[]byte(fileBNewLines),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 7, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 6
+			[]byte(fileBNewLines),
+			[]byte(fileA),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 1, DelLines: 7, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 7
+			[]byte(fileBDelete),
+			[]byte(fileA),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 1, DelLines: 0, Inserted: true, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 1, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("TEST %d/%d", i+1, len(tests)), func(t *testing.T) {
```

Comment:
It'd be nicer to give these tests names and use them here (`t.Run(t.name, func (t *testing.T) {...`.